casepro/msgs/views.py

```diff
@@ -186,7 +187,23 @@ def get_context_data(self, **kwargs):
             user = self.request.user
             page = int(self.request.GET.get('page', 1))
 
+            context['user_id'] = user.id
+
             search = self.derive_search()
+
+            # this is a refresh of messages
+            if self.request.GET.get('last_refresh', None):
+                new_messages = Message.search(org, user, search)
+
+                search['last_refresh'] = self.request.GET['last_refresh']
+                del search['after']
```

Comment:
Example, If a message has a create_date of 2017-01-12 12:00:00 and it was refreshed on 2017-01-13 11:30:00. 

I do a refresh and my last refresh was 2017-01-13 11:29:50, in the search function the queryset will include the message because there was a action after my last_refresh date. 

If after parameter is populated it will try and filter where create_date is later than the after parameter and the message won't be returned. So it is to include old messages with new updates/actions.